langchain/vectorstores/redis.py

```diff
@@ -400,7 +403,14 @@ def from_existing_index(
         except Exception as e:
             raise ValueError(f"Redis failed to connect: {e}")
 
-        return cls(redis_url, index_name, embedding.embed_query)
+        return cls(
```

Comment:
should we pass in rest of kwargs? init passes those through to redis client

Reply:
so confusing where redis client is initialized